src/Symfony/Component/DependencyInjection/Compiler/AbstractRecursivePass.php

```diff
@@ -77,6 +78,9 @@ protected function processValue(mixed $value, bool $isRoot = false)
     {
         if (\is_array($value)) {
             foreach ($value as $k => $v) {
+                if (\is_scalar($v ?? true) && $this->skipScalars) {
```

Comment:
the point is that only `ResolveParameterPlaceHoldersPass` should NOT skip that

Reply:
And `ResolveEnvPlaceholdersPass` also :)